ArrayOperations/Arrays.cs

```diff
@@ -0,0 +1,324 @@
+using System;
+
+namespace ArrayOperations
+{
+  /// <summary>
+  /// This class contain methods for array operations.
+  /// </summary>
+  public static class Arrays
+  {
+    /// <summary>
+    /// Sort an array using 'Bubble Sort' algorithm.
+    /// </summary>
+    /// <param name="input">Source array.</param>
+    /// <exception cref="ArgumentNullException">The source array is not initialized.</exception>
+    /// <exception cref="ArgumentException">The source array has no elements.</exception>
+    public static int[] Sort(int[] input)
+    {
+      if (input == null)
+      {
+        throw new ArgumentNullException(nameof(input));
+      }
+
+      if (input.Length == 0)
+      {
+        throw new ArgumentException("Value cannot be an empty collection.", nameof(input));
+      }
+
+      if (input.Length == 1)
+      {
+        return  input;
+      }
+
+      var midPoint = input.Length / 2;  
+      
+      var left = new int[midPoint];
+      var right = input.Length % 2 == 0 ? new int[midPoint] : new int[midPoint + 1];
```

Comment:
This is great and it utilizes what was taught in class more effectively.  I'm only bringing this up since you are clearly putting in a lot of effort here.  For a merge sort on very large arrays, it's ideal not to have to do all these new arrays and copies.  Making changes in-place will have better performance.  If you want to make one copy at the top-level you can do that too and then you preserve the input array which is often good.

Reply:
I understand, definitely something to study in the following days.